src/Components/Components/test/OwningComponentTest.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Dynamic;
+using System.Linq;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components
+{
+    public class OwningComponentTest
+    {
+        [Fact]
+        public void CreatesScopeAndService()
+        {
+            var services = new ServiceCollection();
+            services.AddTransient<MyService>();
+
+            var renderer = new TestRenderer(services.BuildServiceProvider());
+
+            var i = 0;
+            var component = new TestComponent(builder =>
+            {
+                if (i % 2 == 0)
+                {
+                    builder.OpenComponent<MyOwningComponent>(1);
+                    builder.CloseComponent();
+                }
+
+                builder.AddContent(2, "--");
+
+                if (i % 2 == 1)
+                {
+                    builder.OpenComponent<MyOwningComponent>(3);
+                    builder.CloseComponent();
+                }
+
+                i++;
+            });
```

Comment:
TBH I think it would be sufficient just to render a single instance of `MyOwningComponent`, observe that it did get the message from DI, then explicitly call its `Dispose` method and observe that it did dispose the underlying service. I'm not sure we get much more by simulating a higher-level component that creates and destroys them.

Reply:
ya that would be simpler 👍